crates/driver/src/domain/competition/solution/fee.rs

```diff
@@ -0,0 +1,561 @@
+//! Applies the protocol fee to the solution received from the solver.
+//!
+//! Protocol fee is applied by increasing already existing fee determined by the
+//! solver.
+
+use {
+    super::trade::{Fee, Fulfillment, InvalidExecutedAmount},
+    crate::domain::{
+        competition::{
+            order,
+            order::{FeePolicy, Side},
+        },
+        eth,
+    },
+};
+
+impl Fulfillment {
+    pub fn add_protocol_fee(
+        &self,
+        uniform_sell_price: eth::U256,
+        uniform_buy_price: eth::U256,
```

Comment:
nit: a small helper `ClearingPrice` helper type with fields buy and sell would make this nicer to read.

Reply:
Added.